pkg/actions/fargate/create.go

```diff
@@ -24,37 +25,73 @@ func (m *Manager) Create() error {
 		return fmt.Errorf("Fargate is not supported for this cluster version. Please update the cluster to be at least eks.%d", fargate.MinPlatformVersion)
 	}
 
-	stackManager := ctl.NewStackManager(cfg)
-
-	hasClusterStack, err := stackManager.HasClusterStack()
+	hasClusterStack, err := m.stackManager.HasClusterStack()
 	if err != nil {
 		return errors.Wrap(err, "couldn't check stack")
 	}
 
-	if !hasClusterStack {
-		if err := EnsureUnownedClusterReadyForFargate(cfg, ctl.Provider, stackManager); err != nil {
-			return errors.Wrap(err, "couldn't ensure unowned cluster is ready for fargate")
-		}
-	} else {
-		if err := ctl.LoadClusterIntoSpec(cfg); err != nil {
-			return errors.Wrap(err, "couldn't load cluster into spec")
+	fargateRoleNeeded := false
+
+	for _, profile := range cfg.FargateProfiles {
+		if profile.PodExecutionRoleARN == "" {
+			fargateRoleNeeded = true
+			break
 		}
 	}
 
-	if !api.IsSetAndNonEmptyString(cfg.IAM.FargatePodExecutionRoleARN) {
-		// Read back the default Fargate pod execution role ARN from CloudFormation:
-		if err := ctl.NewStackManager(cfg).RefreshFargatePodExecutionRoleARN(); err != nil {
-			return errors.Wrap(err, "couldn't refresh role arn")
+	if fargateRoleNeeded {
+		if hasClusterStack {
+			exists, err := m.fargateRoleExistsOnClusterStack()
```

Comment:
This adds yet another DescribeStacks call, which is not ideal. I feel that we have like 50 different ways of doing very similar things around stacks?

Is there a way to make `HasClusterStack` on line 28 return something more useful than just a bool? Maybe the actual stack which we can then use here, checking for `!nil` or `!ErrStackNotFound` or whatever?

Reply:
that sounds reasonable, I will do a refactor

Reply:
done